OrbitCore/PathTest.cpp

```diff
@@ -96,3 +97,151 @@ TEST(Path, JoinPathPartsAbsoluteRootBackslash) {
 #endif
   EXPECT_EQ(actual, expected);
 }
+
+TEST(Path, FileNamesAndExtensions) {
+#ifdef _WIN32
+  std::string abs_input[] = {"C:\\some_abs_path\\file.txt", "C:\\dir.with.dots\\file_without_ext"};
+  std::string abs_stripped[] = {"C:\\some_abs_path\\file", "C:\\dir.with.dots\\file_without_ext"};
+#else
+  std::string abs_input[] = {"/some_abs_path/file.txt", "/dir.with.dots/file_without_ext"};
+  std::string abs_stripped[] = {"/some_abs_path/file", "/dir.with.dots/file_without_ext"};
+#endif
+  EXPECT_EQ(Path::GetExtension(abs_input[0]), ".txt");
+  EXPECT_EQ(Path::StripExtension(abs_input[0]), abs_stripped[0]);
+  EXPECT_EQ(Path::GetFileName(abs_input[0]), "file.txt");
+
+  EXPECT_EQ(Path::GetExtension(abs_input[1]), "");
+  EXPECT_EQ(Path::StripExtension(abs_input[1]), abs_stripped[1]);
+  EXPECT_EQ(Path::GetFileName(abs_input[1]), "file_without_ext");
+
+  std::string input = "relative/path/file.txt";
+  EXPECT_EQ(Path::GetExtension(input), ".txt");
+  EXPECT_EQ(Path::StripExtension(input), "relative/path/file");
+  EXPECT_EQ(Path::GetFileName(input), "file.txt");
+
+  input = "simple.file";
+  EXPECT_EQ(Path::GetExtension(input), ".file");
+  EXPECT_EQ(Path::StripExtension(input), "simple");
+  EXPECT_EQ(Path::GetFileName(input), "simple.file");
+
+  input = "no/ext";
+  EXPECT_EQ(Path::GetExtension(input), "");
+  EXPECT_EQ(Path::StripExtension(input), "no/ext");
+  EXPECT_EQ(Path::GetFileName(input), "ext");
+
+  input = "double.ext.txt";
+  EXPECT_EQ(Path::GetExtension(input), ".txt");
+  EXPECT_EQ(Path::StripExtension(input), "double.ext");
+  EXPECT_EQ(Path::StripExtension(Path::StripExtension(input)), "double");
+  EXPECT_EQ(Path::GetFileName(input), "double.ext.txt");
```

Comment:
In general these should also work on linux-style directory separators on windows.. I do not think they should be platform-specific. It is not uncommon to parse a string we recieved via grpc from linux on windows. Or paths from .debug_info section which could be whatever os was using while compiling them.

Reply:
The test actually covers the first part (forward slash on windows) - only the first input is platform specific to check if both platforms handle their respective "native" formats for absolute paths correctly. All other tests are using linux-style directories for both platforms.

I changed one of the tests to cover a mixed-style path for both platforms. This exposed that the `StripExtension` isn't consistent with the other methods - all other methods that return a path convert it to contain only forward-slashes, so I changed `StripExtension` to do that as well, WDYT?